unit-tests/dds/test-librs-connections.py

```diff
@@ -23,53 +23,57 @@
     #
     #############################################################################################
     #
-    test.start( "Multiple cameras disconnection/reconnection" )
-    try:
-        remote.run( 'instance = broadcast_device( d435i, d435i.device_info )', timeout=5 )
-        remote.run( 'instance2 = broadcast_device( d405, d405.device_info )', timeout=5 )
-        
-        #Create context after remote device is ready to test discovery on start-up
-        #Later on, other connections will test discovery using the callback functions
-        context = rs.context( '{"dds-domain":123}' )
-        n_devs = dds.wait_for_devices( context, only_sw_devices )
-        test.check_equal( len(n_devs), 2 )
-        
-        remote.run( 'instance3 = broadcast_device( d455, d455.device_info )', timeout=5 )
-        sleep( 1 ) # Wait for device to be received and built, otherwise wait_for_devices will return immediately with previous devices
-        n_devs = dds.wait_for_devices( context, only_sw_devices )
-        test.check_equal( len(n_devs), 3 )
-        
-        remote.run( 'close_server( instance )', timeout=5 )
+    with test.closure( "Start two devices", on_fail=test.ABORT ):
+        remote.run( 'instance = broadcast_device( d435i, d435i.device_info )' )
+        remote.run( 'instance2 = broadcast_device( d405, d405.device_info )' )
+
+        # Create context after remote device is ready to test discovery on start-up
+        context = rs.context( '{"dds-domain":123,"dds-participant-name":"librs"}' )
+        # The DDS devices take time to be recognized and we just created the context; we
+        # should not see them yet!
+        test.check( len( context.query_devices( only_sw_devices )) != 2 )
```

Comment:
I don't think that we need this check. You don't know the timing and discovery might have happened, especially if working in shared memory.

Reply:
I thought you'd want it based on the original comments.
I agree, but the chances that it'll fail are slim.
Anyway, I commented out the line so it's still explained and there...